source/docs/essentials/extend-the-graphql-schema.md

```diff
@@ -0,0 +1,443 @@
++    { name: "ClicksCounters", path: "./my-module/server/modules/clicks-counters" }
+  ]
+};
+```
+
+The `name` key must be unique across your server modules. It is a temporary name
+that is used during a code generation step and has no other usage in the
+application.
+
+The `path` must be a path to your module definition file (created above).
+
+<blockquote class="info">
+  **In depth:** under the hood, Front-Commerce is generating a
+  `.front-commerce/modules.js` file from this configuration which is loaded very
+  early in the server bootstrapping process. The name is used to import the
+  correct module and export it in an homogeneous list, used by the
+  `withGraphQLApi` express middleware.
+</blockquote>
+
+**Congratulations!** You now have a new custom module ready to enhance the data
+exposed in the GraphQL middleware.
+
+## Implement module’s features
+
+All the wiring is now done and it is time to develop the features of this
+module. In GraphQL a good practice is to start thinking about the schema, from a
+business domain standpoint.
+
+It is important to name things with a language shared by the team and prevent
+exposing implementation details (ids, different names…) as much as possible. We
+recommend the reading of the GraphQL documentation page
+[Thinking in Graphs](https://graphql.github.io/learn/thinking-in-graphs/).
+
+Let’s add the code to expose a counter field and a mutation in our graph.
+
+### Define the schema
+
+Front-Commerce contains all the tooling to allow you to describe your schema
+using the expressive
+[GraphQL Schema Definition Language (SDL)](https://graphql.org/learn/schema/#type-language).
+
+Create a `my-module/server/modules/clicks-counters/schema.gql` file to contain
+the schema matching our use case.
+
+We would like to:
+
+1. add a `clicksCounter` field to the existing `Product` type
+2. add an `incrementProductCounter` mutation
+
+It could look like so:
+
+```graphql
+# my-module/server/modules/clicks-counters/schema.gql
+extend type Product {
+  clicksCounter: Int
+}
+
+extend type Mutation {
+  incrementProductCounter(sku: String!, incrementValue: Int): MutationSuccess
+}
+```
+
+<blockquote class="info">
+  The `Product` type is part of Front-Commerce’s `Magento2` module. The
+  `MutationSuccess` type used as the mutation response type is part of
+  Front-Commerce’s core.
+</blockquote>
+
+Then expose the schema using the `typeDefs` key of the module definition. Update
+the `my-module/server/modules/clicks-counters/index.js` entrypoint:
+
+```diff
+// my-module/server/modules/clicks-counters/index.js
++import typeDefs from "./schema.gql";
++
+export default {
+-  namespace: "ClicksCounters"
++  namespace: "ClicksCounters",
++  typeDefs: typeDefs
+};
+```
+
+<blockquote class="info">
+  Webpack is in fact taking care of converting SDL into a Javascript object
+  using an appropriate loader.
+</blockquote>
+
+**Congratulations again!** You should now be able to see these new fields and
```

Comment:
Maybe I'd add a mention that no restart is needed here because the server is hot reloaded and it's **awesome**.